src/components/Switch/Switch.tsx

```diff
@@ -0,0 +1,25 @@
+import { FC } from 'react';
+import { StyledSwitch } from './Switch.style';
+import { ThemeColor, ThemeElementAppearance, ThemeElementSize } from '../../types.ts';
+
+type SwitchProps = {
+	color?: ThemeColor;
+	onClick: () => void;
+	appearance?: ThemeElementAppearance;
+	size?: ThemeElementSize;
+}
+
+const Switch: FC<SwitchProps> = ({
+	color = 'primary',
+	onClick,
+	appearance = 'solid',
+	size = 'md'
+}) => {

```

Comment:
How do we know if the switch is on or off? It should probably have a boolean state value or something like that, which gets toggled on click. It's not enough for it to _look_ on or off, because its value would be used somewhere in the application.